tests/python/contrib/test_hexagon/infrastructure.py

```diff
@@ -18,36 +18,43 @@
 """ Hexagon testing infrastructure """
 
 import tvm
+from tvm import te
 import numpy
 
 
 def ceildiv(o, d):
     return tvm.tir.floordiv(o + d - 1, d)
 
 
-def get_packed_activation_layout(shape_nhwc, block_shape, packed_C=True):
+def get_block_shape():
+    return 8, 8, 32
+
+
+def get_filter_block_shape():
+    return 8, 32, 4
+
+
+def get_packed_shape(shape_nhwc):
     assert len(shape_nhwc) == 4
     shape = [shape_nhwc[0]]
+    block_shape = get_block_shape()
     off_h, off_w, off_c = block_shape
     shape.append(ceildiv(shape_nhwc[1], off_h))
     shape.append(ceildiv(shape_nhwc[2], off_w))
-    if packed_C:
-        shape.append(ceildiv(shape_nhwc[3], off_c))
-        shape.extend(block_shape)
-    else:
-        shape.extend([off_h, off_w, shape_nhwc[3]])
+    shape.append(ceildiv(shape_nhwc[3], off_c))
```

Comment:
In this code we handle `ceildiv` as-is, but in the `get_packed_filter_shape` function (below) we cast its result to a Python `int` type.

Is there a good reason for doing it two different ways?  

Specifically: casting to `int` works when the runtime return type of `ceildiv` is `tvm.tir.expr.IntImm`.  But I'm wondering if we're okay with this test making that assumption. I.e., are we testing everything we _intend_ to test when we assume that the exact shapes are known at this point in time?

Reply:
Indeed, casting to `int` is a means of converting from `tvm.tir.expr.IntImm` to `int`.  I am not sure why it works differently for `get_packed_shape` (no cast) vs. `get_packed_filter_shape` (requires cast).  Also not sure about your question RE knowing exact shapes.  More investigation required.

NOTE:  I did change `get_packed_filter_shape` to take a shape instead of a list of dimensions to match the API of `get_packed_shape`.